### ISSUE TITLE
Slowdown interacting with denite

### ISSUE THREAD

**Issue by JCavallo** [Shougo/deoplete.nvim#996]
# Problem summary

This is a follow-up on issue https://github.com/Shougo/denite.nvim/issues/676 that I initially thought was caused by the denite plugin.

The problem is a slowdown when starting the denite window, which happens randomly after some time using neovim.

When debugging, I noticed that what I assume is the deoplete channel sends the following rpc calls :

```
2019-07-22 14:23:49,237 [DEBUG @ msgpack_stream.py:send:33] 8816 - sent [0, 7475, 'nvim_call_function', ('deoplete#custom#_get_option', ['smart_case'])]
2019-07-22 14:23:49,237 [DEBUG @ async_session.py:_on_response:100] 8816 - received response: None, 1
2019-07-22 14:23:49,238 [DEBUG @ msgpack_stream.py:send:33] 8816 - sent [0, 7476, 'nvim_eval', ('g:',)]

...

Hangs up for 15 seconds, then :

2019-07-22 14:24:04,804 [DEBUG @ msgpack_stream.py:_on_data:60] 8816 - received message: [1, 7476, None, { HUGE BLOB }]
```

So I have two questions.

I thought I had disabled deoplete for the denite / denite-filter buffers using

```
autocmd FileType denite
       \ call deoplete#custom#buffer_option('auto_complete', v:false)
```

Did I do it wrong? Is there a way to check whether deoplete is effectively active for the denite-buffer?

Second, are the calls above normal?

## Environment Information (Required!)

 * denite version (SHA1): b6a0c7d

 * OS: Ubuntu 18.04

 * Vim/neovim version: NVIM v0.4.0-1016-g6f27f5ef9

 * `:checkhealth` or `:CheckHealth` result(neovim only):

```
health#deoplete#check
========================================================================
## deoplete.nvim
  - OK: exists("v:t_list") was successful
  - OK: has("timers") was successful
  - OK: has("python3") was successful
  - OK: Python3.6.1+ was successful
  - INFO: If you're still having problems, try the following commands:
    $ export NVIM_PYTHON_LOG_FILE=/tmp/log
    $ export NVIM_PYTHON_LOG_LEVEL=DEBUG
    $ nvim
    $ cat /tmp/log_{PID}
    and then create an issue on github

health#denite#check
========================================================================
## denite.nvim
  - OK: has("python3") was successful
  - OK: Python3.6.1+ was successful

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: RelWithDebInfo

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=\177
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $COLORTERM='truecolor'

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: xclip

## Python 2 provider (optional)
  - INFO: pyenv: Path: /home/giovanni/.pyenv/libexec/pyenv
  - INFO: pyenv: $PYENV_ROOT is not set. Infer from `pyenv root`.
  - INFO: pyenv: Root: /home/giovanni/.pyenv
  - INFO: Using: g:python_host_prog = "/home/giovanni/.pyenv/versions/neovim2/bin/python"
  - WARNING: $VIRTUAL_ENV exists but appears to be inactive. This could lead to unexpected results.
    - ADVICE:
      - If you are using Zsh, see: http://vi.stackexchange.com/a/7654
  - INFO: Executable: /home/giovanni/.pyenv/versions/neovim2/bin/python
  - INFO: Python version: 2.7.15
  - INFO: pynvim version: 0.3.2
  - OK: Latest pynvim is installed.

## Python 3 provider (optional)
  - INFO: pyenv: Path: /home/giovanni/.pyenv/libexec/pyenv
  - INFO: pyenv: $PYENV_ROOT is not set. Infer from `pyenv root`.
  - INFO: pyenv: Root: /home/giovanni/.pyenv
  - INFO: Using: g:python3_host_prog = "/home/giovanni/.pyenv/versions/neovim3/bin/python"
  - WARNING: $VIRTUAL_ENV exists but appears to be inactive. This could lead to unexpected results.
    - ADVICE:
      - If you are using Zsh, see: http://vi.stackexchange.com/a/7654
  - INFO: Executable: /home/giovanni/.pyenv/versions/neovim3/bin/python
  - INFO: Python version: 3.7.1
  - INFO: pynvim version: 0.3.2
  - OK: Latest pynvim is installed.

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help |g:ruby_host_prog| for non-standard gem installations.

## Node.js provider (optional)
  - INFO: Node.js: v8.16.0
  - WARNING: Missing "neovim" npm (or yarn) package.
    - ADVICE:
      - Run in shell: npm install -g neovim
      - Run in shell (if you use yarn): yarn global add neovim
```

**Comment by Shougo** [comment]
It does not work.

You must disable deoplete in denite-filter instead of denite.

Please read the FAQ.

```vim
Q: I want to disable deoplete auto completion in denite filter window.

A:
>
	autocmd FileType denite-filter call s:denite_filter_my_settings()
	function! s:denite_filter_my_settings() abort
	  call deoplete#custom#buffer_option('auto_complete', v:false)
	endfunction
```

**Comment by Shougo** [comment]
I have read the log.

Hm.   Context vars seems slow.
I should fix the problem instead.

But some sources use the feature.

**Comment by JCavallo** [comment]
I disabled it in both cases, I just pasted if for the 'denite" filetype.

Indeed I thought I could use the following pattern, since denite / denite-filter have their own filtetypes:

```
Q: I want to disable the auto completion for certain filetypes.

A: Please use |deoplete-options-auto_complete|. >

       autocmd FileType tex
       \ call deoplete#custom#buffer_option('auto_complete', v:false)
```

I'll try using the way you wrote above. Also, I'll try to narrow down what could take so much time (since it takes 100% CPU for those 15 seconds, there may be a problem somewhere in another plugin).
